dags/test_helpers/helpers.py

```diff
@@ -28,20 +33,24 @@ def db_context():
     teardown_test_db_context(context)
 
 
-def execute_dag(dag_id: str, execution_date: str, environment: dict = {}):
+def execute_dag(dag_id: str, execution_date: str, dag_config: dict = {}):
     """Execute a DAG in a specific date this process wait for DAG run or fail to continue"""
-    env = R.merge(os.environ.copy(), environment)
-    print(env)
+
+    subprocess.Popen(["airflow", "dags", "delete", dag_id, "-y"])
```

Comment:
We have to delete everything related to the DAG from the database before running it. Otherwise, the config is overwritten with what has be used in the first run [because of this bug](https://github.com/apache/airflow/issues/15887) (which prevents us from passing in our test db)